### PR TITLE
update mesh section

### PR DESCRIPTION
**Pull request by mbartling**
Moved bulk of mesh section to top level. Simplified api page to look like other api docs. Need to move examples to hosted repos in dev site

@sg- There are a few examples that are inlined right now and I think these should be hosted somewhere as stand alone examples. Where would you like me to put them so you guys can have control over them?

### CODE REVIEW

**Comment by sg-** [comment]
the only inline should be from mesh-minimal-example

**Comment by mbartling** [comment]
Sorry, let me rephrase. From a users' perspective, the main code will still appear inlined but the actual code will reside in a separate repo. This is preferable to having an example embedded directly in the md file
